go/simple/v1/ampli/ampli.go

```diff
@@ -0,0 +1,385 @@
+			"intMax10": stronglyTypedEvent.IntMax10,
+		},
+	}
+}
+
+// EventNoProperties
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20No%20Properties
+//
+// Event with no properties description
+//
+// Owner: Test codegen.
+type EventNoProperties struct {
+	amplitude.Event
+}
+
+func (stronglyTypedEvent EventNoProperties) ToEvent() amplitude.Event {
+	return amplitude.Event{
+		EventType: "Event No Properties",
+	}
+}
+
+// EventObjectTypes
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20Object%20Types
+//
+// Event with Object and Object Array
+//
+// Owner: Test codegen
+//
+// Params:
+//	- RequiredObject: property object type
+//	- RequiredObjectArray: property object array type
+type EventObjectTypes struct {
+	RequiredObject      interface{}
+	RequiredObjectArray []interface{}
+}
+
+func (stronglyTypedEvent EventObjectTypes) ToEvent() amplitude.Event {
+	return amplitude.Event{
+		EventType: "Event Object Types",
+		EventProperties: map[string]interface{}{
+			"requiredObject":      stronglyTypedEvent.RequiredObject,
+			"requiredObjectArray": stronglyTypedEvent.RequiredObjectArray,
+		},
+	}
+}
+
+type Ampli struct {
+	Disabled bool
+	Client   amplitude.Client
+}
+
+// Load initializes the Ampli wrapper.
+// Call once when your application starts.
+func (a *Ampli) Load(options LoadOptions) {
+	a.Disabled = options.Disabled
+
+	if a.Client != nil {
+		log.Default().Printf("Warn: Ampli is already initialized. Ampli.load() should be called once at application start up.")
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration,
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Client.APIKey or APIKey[options.Environment], " +
+			"or options.Client.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration.Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	return !a.Disabled
```

Comment:
`a.Client != nil`?

Reply:
updated to `return !a.Disabled && a.Client != nil`